dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java

```diff
@@ -1377,7 +1377,8 @@ public void updateTaskDefinitionResources(TaskDefinition taskDefinition) {
                 ResourceInfo mainJar = JSONUtils.parseObject(
                         JSONUtils.toJsonString(mainJarObj),
                         ResourceInfo.class);
-                ResourceInfo resourceInfo = updateResourceInfo(taskDefinition.getId(), mainJar);
+                ResourceInfo resourceInfo =
+                        updateResourceInfo(taskDefinitionMapper.queryByCode(taskDefinition.getCode()).getId(), mainJar);
```

Comment:
Why not use `taskDefinition.getId()` directly?

Reply:
The type of taskDefinition here is `taskDefinitionLog`, resource binds with `taskDefinition`, rather `taskDefinitionLog`.

Reply:
> The type of taskDefinition here is `taskDefinitionLog`, however, resource binds with `taskDefinition`.

Get it. What about expand `updateResourceInfo` with taskCode and version? like 
`protected ResourceInfo updateResourceInfo(int task_id, ResourceInfo res)`,  and need to add some comments for it.

Reply:
> What about expand updateResourceInfo with taskCode and version?

May i ask why we have to add `taskVersion`?

Reply:
> > What about expand updateResourceInfo with taskCode and version?
> 
> May i ask why we have to add `taskVersion`?

Yes, I think we need to add code and version, `taskDefinitionMapper.queryByCode(taskDefinition.getCode())` means query the current using version, but current version maybe is not queal `taskDefinition.version`.

Reply:
> but current version maybe is not equal taskDefinition.version.

Current version may be different, but they have the same code, which is equal to `taskDefinition.code`, so that we can query the right resource task relation.

Reply:
> but current version maybe is not equal taskDefinition.version.

Current version may be different, but they have the same code, which is equal to `taskDefinition.code`, so that we can query the right resource task relation.